processing/db_handler.py

```diff
@@ -196,6 +197,181 @@ def get_filename_checksum_lookup(self, conn):
+                    'u.data_file, u.upload_token, '
+                    'u.upload_comment, u.upload_type_id '
+                    'FROM input_interface.data_upload_log u '
+                    'LEFT JOIN '
+                    'input_interface.data_upload_file_xfer_log x '
+                    'ON u.log_id = x.upload_log_id '
+                    'WHERE '
+                    'u.upload_type_id IN (4, 7) '
+                    'AND x.xfer_end_log_timestamp IS NOT NULL '
+                    'AND u.upload_token = %(uuid)s')
+        params = {'uuid': uuid}
+        with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            cursor.execute(query, params)
+            new_data_upload = cursor.fetchall()
+        return new_data_upload
+
+    def get_new_data_upload_log(self,
+                                conn,
+                                qaqc_processor_source,
+                                is_qaqc_processor,
+                                uuid=None):
+        query_base = SQL('SELECT u.log_id, u.site_id, '
+                         'u.data_file, u.upload_token, '
+                         'u.upload_comment, u.upload_type_id '
+                         'FROM input_interface.data_upload_log u '
+                         'LEFT JOIN qaqc.processing_log p '
+                         'ON u.log_id = p.upload_id '
+                         'LEFT JOIN '
+                         'input_interface.data_upload_file_xfer_log x '
+                         'ON u.log_id = x.upload_log_id '
+                         'LEFT JOIN input_interface.data_source_type s '
+                         'ON u.upload_source_id = s.source_id '
+                         'WHERE p.log_id IS NULL '
+                         'AND u.upload_type_id IN (4, 7) '
+                         'AND x.xfer_end_log_timestamp IS NOT NULL ')
+        if is_qaqc_processor:
+            query_add_1 = SQL('AND s.source = %(qaqc_processor_source)s')
+        else:
+            query_add_1 = SQL('AND s.source != %(qaqc_processor_source)s')
+        query = Composed([query_base, query_add_1])
+        params = {'qaqc_processor_source': qaqc_processor_source}
+        if uuid:
+            query_add_2 = SQL('AND u.upload_token = %(uuid)s')
+            query = Composed([query, query_add_2])
+            params['uuid'] = uuid
+        with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            cursor.execute(query, params)
+            new_data_upload = cursor.fetchall()
+        return new_data_upload
+
+    def get_latest_run_with_uuid(self, conn, uuid):
+        query = SQL('SELECT u.log_id, p.process_timestamp '
+                    'FROM input_interface.data_upload_log u '
+                    'LEFT JOIN qaqc.processing_log p '
+                    'ON u.log_id = p.upload_id '
+                    'WHERE u.upload_token = %(uuid)s '
+                    'ORDER BY p.process_timestamp DESC '
+                    'LIMIT 1')
```

Comment:
This query is not efficient, probably better to filter based on the latest process_timestamp and UUID, and then do the left join.